core/pta/bcm/elog.c

```diff
@@ -89,12 +89,16 @@ static TEE_Result pta_elog_load_nitro_fw(uint32_t param_types,
 		return TEE_ERROR_ACCESS_DENIED;
 	}
 
-	src_vaddr = (vaddr_t)phys_to_virt((uintptr_t)src_paddr + offset,
-					  MEM_AREA_RAM_SEC);
-
 	buf = params[0].memref.buffer;
 	sz = params[0].memref.size;
 
+	src_vaddr = (vaddr_t)phys_to_virt((uintptr_t)src_paddr + offset,
+					  MEM_AREA_RAM_SEC, sz);
+	if (!src_vaddr) {
+		EMSG("Not enough memory mapped");
+		return TEE_ERROR_OUT_OF_MEMORY;
```

Comment:
`TEE_ERROR_OUT_OF_MEMORY` return value could be misinterpreted. Prefer `TEE_ERROR_BAD_PARAMETERS` IMO.